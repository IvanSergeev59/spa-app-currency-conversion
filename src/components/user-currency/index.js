import UserCurrencyContainer from "./user-currency";

export default UserCurrencyContainer