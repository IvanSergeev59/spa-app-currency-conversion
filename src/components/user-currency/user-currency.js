import {React, Component} from "react";
import { Form } from "react-bootstrap";
import { connect } from "react-redux";
import { changeCurrency, currenciesHideDefaultCurrencies} from "../../actions";



 const UserCurrency = ({userChangeCurrency, hideDefaultCurrencies}) => {  
     const hideDefaultCurrency =() => {
         return hideDefaultCurrencies()
     }   
    const onUserChangeCurrency = (event) => {
        const item = event.target.value;  
        hideDefaultCurrency()      
        return userChangeCurrency(item)
    }
        return (
          <div className="currency-list currency-list-user">
            <p className="currency-list-_p" >Ваша валюта:</p>
            <Form.Select
            onChange = {onUserChangeCurrency} aria-label="Default select example">
              
                <option value="USD">USD</option>
                <option value="EURO">EURO</option>
                <option value="RUB">RUB</option>
            </Form.Select>
          </div>
        )
   }


export class UserCurrencyContainer extends Component  {
    render () {
        const {userChangeCurrency, hideDefaultCurrencies} = this.props
        return (
        <UserCurrency userChangeCurrency={userChangeCurrency} hideDefaultCurrencies={hideDefaultCurrencies}/>
        )
    }
}

const mapStateToProps = ({userChangeCurrency}) => {
    return {userChangeCurrency}
} 

const mapDispatchToProps = (dispatch, action) => {
    return {       
        userChangeCurrency: (item) => dispatch(changeCurrency(item)),
        hideDefaultCurrencies: () => dispatch(currenciesHideDefaultCurrencies())
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(UserCurrencyContainer)
