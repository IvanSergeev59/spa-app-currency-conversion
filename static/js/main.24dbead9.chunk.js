(this["webpackJsonpspa-app-currency-conversion"]=this["webpackJsonpspa-app-currency-conversion"]||[]).push([[0],{34:function(e,r,t){},36:function(e,r,t){},37:function(e,r,t){"use strict";t.r(r);var c=t(0),n=t(9),s=t.n(n),a=t(8),u=(t(34),t(1)),l=function(){return Object(u.jsx)("header",{className:"header",children:Object(u.jsx)("h2",{className:"h2",children:"\u041a\u043e\u043d\u0432\u0435\u0440\u0442\u0430\u0446\u0438\u044f \u0432\u0430\u043b\u044e\u0442"})})},i=t(17),o=t(18),j=t(24),d=t(22),b=t(40),O=(t(36),function(e){e.currentUsd,e.totalUsd;var r=e.sumTotalUsd,t=e.sumUsd,c=e.userCurrency;return Object(u.jsxs)("div",{className:"currency-list",children:[Object(u.jsx)("div",{className:"form",children:Object(u.jsxs)(b.a.Group,{className:"mb-3",controlId:"formBasicEmail",children:[Object(u.jsxs)(b.a.Label,{children:[c," in RUB"]}),Object(u.jsx)(b.a.Control,{onChange:function(e){var t=e.target.value;return r(t)},type:"number",placeholder:"\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u043a\u043e\u043b\u0438\u0447\u0435\u0441\u0442\u0432\u043e"})]})}),Object(u.jsx)("p",{className:"result-block",children:"\u0418\u0442\u043e\u0433\u043e:"}),Object(u.jsx)("p",{className:"result-field",children:t})]})}),h=function(e){var r=e.userChangeCurrency;return Object(u.jsxs)("div",{class:"currency-list currency-list-user",children:[Object(u.jsx)("p",{class:"currency-list-_p",children:"\u0412\u0430\u0448\u0430 \u0432\u0430\u043b\u044e\u0442\u0430:"}),Object(u.jsxs)(b.a.Select,{onChange:function(e){var t=e.target.value;return console.log(t),r(t)},"aria-label":"Default select example",children:[Object(u.jsx)("option",{value:"USD",children:"USD"}),Object(u.jsx)("option",{value:"EURO",children:"EURO"}),Object(u.jsx)("option",{value:"RUB",children:"RUB"})]})]})},p=function(e){Object(j.a)(t,e);var r=Object(d.a)(t);function t(){return Object(i.a)(this,t),r.apply(this,arguments)}return Object(o.a)(t,[{key:"render",value:function(){var e=this.props,r=e.sumTotalUsd,t=e.sumUsd,n=e.userCurrency,s=e.userChangeCurrency;return Object(u.jsxs)(c.Fragment,{children:[Object(u.jsx)(h,{userChangeCurrency:s}),Object(u.jsx)(O,{sumUsd:t,sumTotalUsd:r,userCurrency:n})]})}}]),t}(c.Component),U=Object(a.b)((function(e){var r=e.currentUsd,t=e.currentRUB,c=e.totalUsd,n=e.userCurrency,s=e.currentEURO,a=0;switch(n){case"USD":a=r*c;break;case"EURO":a=s*c;break;case"RUB":a=t*c;break;default:a=0}return{sumUsd:a,userCurrency:n}}),(function(e,r){return{sumTotalUsd:function(r){return e(function(e){return{type:"SUM_TOTAL",payload:e}}(r))},userChangeCurrency:function(r){return e(function(e){return{type:"CHANGE_CURRENCY",payload:e}}(r))}}}))(p),m=function(){return Object(u.jsx)(c.Fragment,{children:Object(u.jsx)(U,{})})},y=function(){return Object(u.jsxs)("div",{className:"App",children:[Object(u.jsx)(l,{}),Object(u.jsx)(m,{})]})},x=t(23),f=t(2),C={currentUsd:"70",totalUsd:"0",sumUsd:"0",currentEURO:"90",currentRUB:"1",userCurrency:"USD"},v=function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:C,r=arguments.length>1?arguments[1]:void 0;switch(r.type){case"SUM_TOTAL":return Object(f.a)(Object(f.a)({},e),{},{totalUsd:r.payload});case"CHANGE_CURRENCY":return Object(f.a)(Object(f.a)({},e),{},{userCurrency:r.payload});default:return Object(f.a)({},e)}},R=t(20),g=Object(R.a)(v);s.a.render(Object(u.jsx)(a.a,{store:g,children:Object(u.jsx)(x.a,{children:Object(u.jsx)(y,{})})}),document.getElementById("root"))}},[[37,1,2]]]);
//# sourceMappingURL=main.24dbead9.chunk.js.map