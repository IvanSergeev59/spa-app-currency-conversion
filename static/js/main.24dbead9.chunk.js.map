{"version":3,"sources":["components/header/index.js","components/header/header.js","components/currency-list/currency-list.js","actions/index.js","components/pages/homepage.js","app.js","reducers/index.js","store.js","index.js"],"names":["Header","className","CurrencyList","currentUsd","totalUsd","sumTotalUsd","sumUsd","userCurrency","Form","Group","controlId","Label","Control","onChange","event","item","target","value","type","placeholder","UserCurrency","userChangeCurrency","class","Select","console","log","CurrencyListContainer","this","props","Component","connect","currentRUB","currentEURO","dispatch","action","payload","sumTotal","changeCurrency","HomePage","App","initialState","reducer","state","store","createStore","ReactDOM","render","document","getElementById"],"mappings":"mPAEeA,ECIA,WACX,OACI,wBAAQC,UAAU,SAAlB,SACI,oBAAIA,UAAU,KAAd,kH,wCCHNC,G,MAAe,SAAC,GAA+D,EAA7DC,WAA6D,EAAjDC,SAAkD,IAAxCC,EAAuC,EAAvCA,YAAaC,EAA0B,EAA1BA,OAAQC,EAAkB,EAAlBA,aAO/D,OAEI,sBAAKN,UAAU,gBAAf,UACI,qBAAKA,UAAU,OAAf,SACI,eAACO,EAAA,EAAKC,MAAN,CAAYR,UAAU,OAAOS,UAAU,iBAAvC,UACI,eAACF,EAAA,EAAKG,MAAN,WAAaJ,EAAb,aACA,cAACC,EAAA,EAAKI,QAAN,CACAC,SAbI,SAACC,GACjB,IAAMC,EAAMD,EAAME,OAAOC,MACzB,OAAOZ,EAAYU,IAWgBG,KAAK,SAASC,YAAY,iHAKzD,mBAAGlB,UAAU,eAAb,6CACA,mBAAGA,UAAU,eAAb,SAA6BK,SAMnCc,EAAe,SAAC,GAA0B,IAAzBC,EAAwB,EAAxBA,mBAMf,OACE,sBAAKC,MAAM,mCAAX,UACE,mBAAGA,MAAM,mBAAT,4EACA,eAACd,EAAA,EAAKe,OAAN,CACAV,SATM,SAACC,GACX,IAAMC,EAAOD,EAAME,OAAOC,MAE1B,OADAO,QAAQC,IAAIV,GACLM,EAAmBN,IAMH,aAAW,yBAD9B,UAGI,wBAAQE,MAAM,MAAd,iBACA,wBAAQA,MAAM,OAAd,kBACA,wBAAQA,MAAM,MAAd,wBAMVS,E,4JAEF,WACI,MAAgEC,KAAKC,MAA9DvB,EAAP,EAAOA,YAAaC,EAApB,EAAoBA,OAAQC,EAA5B,EAA4BA,aAAcc,EAA1C,EAA0CA,mBAC1C,OACI,eAAC,WAAD,WACI,cAAC,EAAD,CAAcA,mBAAoBA,IAClC,cAAC,EAAD,CAAcf,OAAQA,EAAQD,YAAaA,EAAaE,aAAcA,W,GAPlDsB,aAuCrBC,eA1BS,SAAC,GAAmE,IAAlE3B,EAAiE,EAAjEA,WAAY4B,EAAqD,EAArDA,WAAY3B,EAAyC,EAAzCA,SAAUG,EAA+B,EAA/BA,aAAcyB,EAAiB,EAAjBA,YAClE1B,EAAS,EACb,OAAQC,GACJ,IAAK,MACDD,EAASH,EAAWC,EACpB,MACJ,IAAK,OACAE,EAAS0B,EAAY5B,EACrB,MACL,IAAK,MACAE,EAASyB,EAAW3B,EACpB,MACL,QACIE,EAAS,EAGjB,MAAO,CAACA,SAAQC,mBAGO,SAAC0B,EAAUC,GAClC,MAAO,CACH7B,YAAa,SAACU,GAAD,OAAUkB,ECtFd,SAAClB,GACd,MAAO,CACHG,KAAM,YACNiB,QAASpB,GDmFuBqB,CAASrB,KACzCM,mBAAoB,SAACN,GAAD,OAAUkB,EChFf,SAAClB,GACpB,MAAO,CACHG,KAAM,kBACNiB,QAASpB,GD6E8BsB,CAAetB,QAI/Ce,CAA6CJ,GEhF7CY,EARE,WACb,OACI,cAAC,WAAD,UACI,cAAC,EAAD,OCKGC,EATH,WACV,OACE,sBAAKtC,UAAU,MAAf,UACK,cAAC,EAAD,IACA,cAAC,EAAD,Q,eCNHuC,EAAe,CACjBrC,WAAW,KACXC,SAAU,IACVE,OAAQ,IACR0B,YAAa,KACbD,WAAY,IACZxB,aAAc,OAoBHkC,EAjBC,WAAiC,IAAhCC,EAA+B,uDAAzBF,EAAcN,EAAW,uCAC5C,OAAQA,EAAOhB,MACX,IAAK,YACD,OAAO,2BACDwB,GADN,IAEGtC,SAAU8B,EAAOC,UAGxB,IAAK,kBACD,OAAO,2BACAO,GADP,IAEInC,aAAc2B,EAAOC,UAE7B,QAAS,OAAO,eAAIO,K,QClBbC,EAFDC,YAAYH,GCK1BI,IAASC,OACP,cAAC,IAAD,CAAUH,MAAOA,EAAjB,SACE,cAAC,IAAD,UACE,cAAC,EAAD,QAGJI,SAASC,eAAe,W","file":"static/js/main.24dbead9.chunk.js","sourcesContent":["import Header from \"./header\";\r\n\r\nexport default Header","import { React, Component } from \"react\";\r\nimport {connect} from 'react-redux';\r\nimport {variant2} from '../../actions'\r\nimport Button from 'react-bootstrap/Button';\r\nimport './header.css'\r\n\r\nconst Header = () => {    \r\n    return (\r\n        <header className=\"header\">\r\n            <h2 className=\"h2\">Конвертация валют</h2>            \r\n        </header>\r\n    )\r\n    \r\n}\r\n\r\n\r\nexport default Header","import {React, Component, Fragment} from \"react\";\r\nimport { Form} from \"react-bootstrap\";\r\nimport { connect } from \"react-redux\";\r\nimport './currency-list.css';\r\nimport { sumTotal, changeCurrency } from \"../../actions\";\r\n\r\nconst CurrencyList = ( {currentUsd, totalUsd, sumTotalUsd, sumUsd, userCurrency}) => { \r\n    const insertTotal = (event) => {        \r\n        const item =event.target.value;\r\n        return sumTotalUsd(item)\r\n    }\r\n\r\n\r\n    return (        \r\n\r\n        <div className=\"currency-list\">\r\n            <div className=\"form\" >\r\n                <Form.Group className=\"mb-3\" controlId=\"formBasicEmail\">\r\n                    <Form.Label>{userCurrency} in RUB</Form.Label>\r\n                    <Form.Control\r\n                    onChange={insertTotal} type=\"number\" placeholder=\"Введите количество\" />\r\n                   \r\n                </Form.Group>\r\n                \r\n            </div>\r\n            <p className=\"result-block\">Итого:</p>\r\n            <p className=\"result-field\">{sumUsd}</p>\r\n        </div>\r\n    )\r\n}\r\n\r\n\r\nconst UserCurrency = ({userChangeCurrency}) => {\r\n    const ololo = (event) => {\r\n        const item = event.target.value;\r\n        console.log(item)\r\n        return userChangeCurrency(item)\r\n    }\r\n        return (\r\n          <div class=\"currency-list currency-list-user\">\r\n            <p class=\"currency-list-_p\" >Ваша валюта:</p>\r\n            <Form.Select\r\n            onChange = {ololo} aria-label=\"Default select example\">\r\n              \r\n                <option value=\"USD\">USD</option>\r\n                <option value=\"EURO\">EURO</option>\r\n                <option value=\"RUB\">RUB</option>\r\n            </Form.Select>\r\n          </div>\r\n        )\r\n}\r\n\r\nclass CurrencyListContainer extends Component {   \r\n\r\n    render () {\r\n        const {sumTotalUsd, sumUsd, userCurrency, userChangeCurrency} = this.props\r\n        return (\r\n            <Fragment>\r\n                <UserCurrency userChangeCurrency={userChangeCurrency}/>\r\n                <CurrencyList sumUsd={sumUsd} sumTotalUsd={sumTotalUsd} userCurrency={userCurrency}/>\r\n            </Fragment>\r\n        )\r\n    }\r\n}\r\n\r\nconst mapStateToProps = ({currentUsd, currentRUB, totalUsd, userCurrency, currentEURO}) => {\r\n    let sumUsd = 0;\r\n    switch (userCurrency) {\r\n        case 'USD':\r\n            sumUsd = currentUsd*totalUsd;\r\n            break\r\n        case 'EURO':\r\n             sumUsd = currentEURO*totalUsd;\r\n             break\r\n        case 'RUB':\r\n             sumUsd = currentRUB*totalUsd;\r\n             break\r\n        default:\r\n            sumUsd = 0\r\n    }\r\n\r\n    return {sumUsd, userCurrency}\r\n} \r\n\r\nconst mapDispatchToProps = (dispatch, action) => {\r\n    return {\r\n        sumTotalUsd: (item) => dispatch(sumTotal(item)),\r\n        userChangeCurrency: (item) => dispatch(changeCurrency(item))\r\n    }\r\n}\r\n\r\nexport default connect(mapStateToProps, mapDispatchToProps)(CurrencyListContainer)","const sumTotal = (item) => {\r\n    return {\r\n        type: 'SUM_TOTAL',\r\n        payload: item\r\n    }\r\n}\r\n\r\nconst changeCurrency = (item) => {\r\n    return {\r\n        type: 'CHANGE_CURRENCY',\r\n        payload: item\r\n    }\r\n}\r\n\r\nexport {sumTotal, changeCurrency}","import {React, Fragment } from \"react\";\r\nimport CurrencyListContainer from \"../currency-list/currency-list\";\r\n\r\nconst HomePage = () => {\r\n    return (\r\n        <Fragment>\r\n            <CurrencyListContainer />\r\n        </Fragment>  \r\n    )\r\n    \r\n}\r\nexport default HomePage","import Header from \"./components/header\";\r\nimport {HomePage} from \"./components/pages\"\r\nconst App = () => {\r\n  return (\r\n    <div className=\"App\">\r\n         <Header />\r\n         <HomePage />  \r\n    </div>\r\n  );\r\n}\r\n\r\nexport default App;\r\n","const initialState = {\r\n    currentUsd:'70',\r\n    totalUsd: '0',\r\n    sumUsd: '0',\r\n    currentEURO: '90',\r\n    currentRUB: '1',\r\n    userCurrency: 'USD'\r\n}\r\n\r\nconst reducer = (state=initialState, action) => {\r\n    switch (action.type) {\r\n        case 'SUM_TOTAL':\r\n            return {\r\n               ...state,\r\n               totalUsd: action.payload\r\n            };\r\n\r\n        case 'CHANGE_CURRENCY':\r\n            return {\r\n                ...state,\r\n                userCurrency: action.payload\r\n            }\r\n        default: return {...state}\r\n    }\r\n}\r\n\r\nexport default reducer\r\n","import reducer from \"./reducers\";\r\nimport {createStore} from 'redux';\r\nconst store = createStore(reducer);\r\n\r\nexport default store","import React from 'react';\r\nimport ReactDOM from 'react-dom';\r\nimport App from './app';\r\nimport { Provider } from \"react-redux\";\r\nimport { BrowserRouter as Router} from \"react-router-dom\";\r\nimport store from './store';\r\n\r\nReactDOM.render(\r\n  <Provider store={store}>\r\n    <Router>\r\n      <App />\r\n    </Router>\r\n  </Provider>,\r\n  document.getElementById('root')\r\n);\r\n"],"sourceRoot":""}