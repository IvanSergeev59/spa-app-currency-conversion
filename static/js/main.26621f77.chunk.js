(this["webpackJsonpspa-app-currency-conversion"]=this["webpackJsonpspa-app-currency-conversion"]||[]).push([[0],{34:function(e,r,n){},41:function(e,r,n){},42:function(e,r,n){},44:function(e,r,n){"use strict";n.r(r);var u=n(0),c=n(17),t=n.n(c),a=n(8),s=n(9),o=n(13),i=n(12),l=n(14),y=(n(34),n(1)),d=function(){return Object(y.jsxs)("header",{className:"header",children:[Object(y.jsx)("h3",{className:"header__h3",children:Object(y.jsx)(l.b,{className:"header__a",to:"/",children:"\u041a\u043e\u043d\u0432\u0435\u0440\u0442\u0435\u0440 \u0432\u0430\u043b\u044e\u0442"})}),Object(y.jsxs)("div",{className:"header__container",children:[Object(y.jsx)(l.b,{className:"header__a",to:"/",children:"\u0413\u043b\u0430\u0432\u043d\u0430\u044f"}),Object(y.jsx)(l.b,{className:"header__a",to:"/current-exchange-rate/",children:"\u041a\u0443\u0440\u0441 \u0432\u0430\u043b\u044e\u0442"})]})]})},p=n(45),j=n(6),b=(n(41),function(e){var r=e.sumTotalUsd,n=e.errorUserWrongInput,u=(e.userSumInputMoney,e.userCurrency),c=e.exchangeCurrency_1,t=e.exchangeCurrency_2,a=e.userSumInputMoney_1,s=e.userSumInputMoney_2;return Object(y.jsxs)("div",{className:"currency-list",children:[Object(y.jsxs)("div",{className:"form",children:[Object(y.jsxs)(p.a.Group,{className:"currency-list-margin",children:[Object(y.jsx)(p.a.Label,{children:"\u041a\u043e\u043b\u0438\u0447\u0435\u0441\u0442\u0432\u043e"}),Object(y.jsx)(p.a.Control,{onChange:function(e){var n=e.target.value;return r(n)},id:"only_num",type:"number",placeholder:"\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u043a\u043e\u043b\u0438\u0447\u0435\u0441\u0442\u0432\u043e"})]}),Object(y.jsx)("p",{className:"currency-list__error",children:n}),Object(y.jsxs)("p",{className:"result-block",children:[u," in ",c,":"]}),Object(y.jsx)("p",{className:"result-field",children:a})]}),Object(y.jsxs)("p",{className:"result-block",children:[u," in ",t,":"]}),Object(y.jsx)("p",{className:"result-field",children:s})]})}),h=function(e){Object(o.a)(n,e);var r=Object(i.a)(n);function n(){return Object(a.a)(this,n),r.apply(this,arguments)}return Object(s.a)(n,[{key:"render",value:function(){var e=this.props,r=e.sumTotalUsd,n=e.userCurrency,u=e.exchangeCurrency_1,c=e.exchangeCurrency_2,t=e.userSumInputMoney_1,a=e.errorUserWrongInput,s=e.userSumInputMoney_2;return Object(y.jsx)(b,{sumTotalUsd:r,errorUserWrongInput:a,exchangeCurrency_1:u,exchangeCurrency_2:c,userCurrency:n,userSumInputMoney_1:t,userSumInputMoney_2:s})}}]),n}(u.Component),C=Object(j.b)((function(e){var r=e.currentCurrencies,n=r.errorUserWrongInput;return{userSumInputMoney_1:r.userSumInputMoney_1,userSumInputMoney_2:r.userSumInputMoney_2,errorUserWrongInput:n,userCurrency:r.userCurrency,exchangeCurrency_1:r.exchangeCurrency_1,exchangeCurrency_2:r.exchangeCurrency_2}}),(function(e,r){return{sumTotalUsd:function(r){return e(function(e){return{type:"SUM_TOTAL",payload:e}}(r))}}}))(h),x=function(){return Object(y.jsx)(u.Fragment,{children:Object(y.jsx)(C,{})})},O=(n(42),Object(j.b)((function(e){var r=e.currentCurrencies,n=r.currentUsdToRub,u=r.currentUsdToEuro,c=r.currentEuroToRub,t=r.currentEuroToUsd,a=r.currentRubToEuro,s=r.currentRubToUsd,o=r.userCurrency,i=r.exchangeCurrency_1,l=r.exchangeCurrency_2;switch(o){case"USD":return{exchangeCurrency_1:i,exchangeCurrency_2:l,currency1:u,currency2:n};case"EURO":return{exchangeCurrency_1:i,exchangeCurrency_2:l,currency1:t,currency2:c};case"RUB":return{exchangeCurrency_1:i,exchangeCurrency_2:l,currency1:s,currency2:a};default:return null}}))((function(e){var r=e.currency1,n=e.currency2,c=e.exchangeCurrency_1,t=e.exchangeCurrency_2;return Object(y.jsxs)(u.Fragment,{children:[Object(y.jsx)("h3",{className:"pages pages__h3",children:"\u0410\u043a\u0442\u0443\u0430\u043b\u044c\u043d\u044b\u0439 \u043a\u0443\u0440\u0441 \u0432\u0430\u043b\u044e\u0442:"}),Object(y.jsxs)("div",{className:"pages__currency-window",children:[Object(y.jsxs)("p",{className:"pages__currency-window_p",children:[c,":"]}),Object(y.jsx)("p",{className:"pages__currency-window_p",children:r})]}),Object(y.jsxs)("div",{className:"pages__currency-window",children:[Object(y.jsxs)("p",{className:"pages__currency-window_p",children:[t,":"]}),Object(y.jsx)("p",{className:"pages__currency-window_p",children:n})]})]})}))),_=function(e){var r=e.userChangeCurrency;return Object(y.jsxs)("div",{className:"currency-list currency-list-user",children:[Object(y.jsx)("p",{className:"currency-list-_p",children:"\u0412\u0430\u0448\u0430 \u0432\u0430\u043b\u044e\u0442\u0430:"}),Object(y.jsxs)(p.a.Select,{onChange:function(e){var n=e.target.value;return r(n)},"aria-label":"Default select example",children:[Object(y.jsx)("option",{value:"USD",children:"USD"}),Object(y.jsx)("option",{value:"EURO",children:"EURO"}),Object(y.jsx)("option",{value:"RUB",children:"RUB"})]})]})},m=function(e){Object(o.a)(n,e);var r=Object(i.a)(n);function n(){return Object(a.a)(this,n),r.apply(this,arguments)}return Object(s.a)(n,[{key:"render",value:function(){var e=this.props.userChangeCurrency;return Object(y.jsx)(_,{userChangeCurrency:e})}}]),n}(u.Component),g=Object(j.b)((function(e){return{userChangeCurrency:e.userChangeCurrency}}),(function(e,r){return{userChangeCurrency:function(r){return e(function(e){return{type:"CHANGE_CURRENCY",payload:e}}(r))}}}))(m),U=n(16),f=n.n(U),v=n(23),R=function(){function e(){var r=this;Object(a.a)(this,e),this.apiBase="https://www.cbr-xml-daily.ru/daily_json.js",this.getCurrencies=Object(v.a)(f.a.mark((function e(){var n,u;return f.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,r.getCurrency();case 2:return n=e.sent,u={currentUsdToRub:n.Valute.USD.Value,currentEuroToRub:n.Valute.EUR.Value},console.log(u),e.abrupt("return",u);case 6:case"end":return e.stop()}}),e)})))}return Object(s.a)(e,[{key:"getCurrency",value:function(){var e=Object(v.a)(f.a.mark((function e(r){var n;return f.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch(this.apiBase);case 2:if((n=e.sent).ok){e.next=5;break}throw new Error("Could not fetch ".concat(this.apiBase)+" , received ".concat(n.status));case 5:return e.abrupt("return",n.json());case 6:case"end":return e.stop()}}),e,this)})));return function(r){return e.apply(this,arguments)}}()}]),e}(),T=R,E=n(3),S=function(e){Object(o.a)(n,e);var r=Object(i.a)(n);function n(){var e;Object(a.a)(this,n);for(var u=arguments.length,c=new Array(u),t=0;t<u;t++)c[t]=arguments[t];return(e=r.call.apply(r,[this].concat(c))).currencyService=new T,e.loadingCurrentCurrency=function(){e.currencyService.getCurrencies().then(e.props.onLoadCurrentCurrency).catch()},e}return Object(s.a)(n,[{key:"componentDidMount",value:function(){this.loadingCurrentCurrency()}},{key:"render",value:function(){return Object(y.jsxs)("div",{className:"App",children:[Object(y.jsx)(d,{}),Object(y.jsx)(g,{}),Object(y.jsxs)(E.c,{children:[Object(y.jsx)(E.a,{path:"/spa-app-currency-conversion/",element:Object(y.jsx)(x,{})}),Object(y.jsx)(E.a,{path:"/*",element:Object(y.jsx)(x,{})}),Object(y.jsx)(E.a,{path:"/current-exchange-rate/",element:Object(y.jsx)(O,{})})]})]})}}]),n}(u.Component),I=Object(j.b)((function(e){return e}),(function(e,r){return{onLoadCurrentCurrency:function(r){return e(function(e){return{type:"UPDATE_CURRENCY",payload:e}}(r))}}}))(S),M=n(2),N=function(e,r){if(void 0===e)return{currentUsdToRub:"70",currentUsdToEuro:"0",currentEuroToRub:"70",currentEurotoUsd:"0",currentRubToEuro:"70",currentRubToUsd:"0",userSumInputMoney_1:"0",userSumInputMoney_2:"0",userCurrency:"USD",exchangeCurrency_1:"EURO",exchangeCurrency_2:"RUB",total:"0",errorUserWrongInput:" "};switch(r.type){case"SUM_TOTAL":return function(e,r){var n=e.currentCurrencies,u=n.userCurrency,c=n.currentUsdToRub,t=n.currentUsdToEuro,a=n.currentEuroToRub,s=n.currentEuroToUsd,o=n.currentRubToEuro,i=n.currentRubToUsd;return/^(0|[1-9]\d*)$/.test(r.payload)&r.payload>=0?"USD"===u?Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userSumInputMoney_1:(r.payload*t).toFixed(4),userSumInputMoney_2:(r.payload*c).toFixed(4),total:r.payload,errorUserWrongInput:" "}):"EURO"===u?Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userSumInputMoney_1:(r.payload*s).toFixed(4),userSumInputMoney_2:(r.payload*a).toFixed(4),total:r.payload,errorUserWrongInput:" "}):Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userSumInputMoney_1:(r.payload*i).toFixed(4),userSumInputMoney_2:(r.payload*o).toFixed(4),total:r.payload,errorUserWrongInput:" "}):Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userSumInputMoney_1:0,userSumInputMoney_2:0,total:0,errorUserWrongInput:"\u041d\u0435\u0432\u0435\u0440\u043d\u043e\u0435 \u0437\u043d\u0430\u0447\u0435\u043d\u0438\u0435"})}(e,r);case"CHANGE_CURRENCY":return function(e,r){return"USD"===r.payload?Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userCurrency:r.payload,exchangeCurrency_1:"EURO",exchangeCurrency_2:"RUB",userSumInputMoney_1:(e.currentCurrencies.total*e.currentCurrencies.currentUsdToEuro).toFixed(4),userSumInputMoney_2:(e.currentCurrencies.total*e.currentCurrencies.currentUsdToRub).toFixed(4)}):"EURO"===r.payload?Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userCurrency:r.payload,exchangeCurrency_1:"USD",exchangeCurrency_2:"RUB",userSumInputMoney_1:(e.currentCurrencies.total*e.currentCurrencies.currentEuroToUsd).toFixed(4),userSumInputMoney_2:(e.currentCurrencies.total*e.currentCurrencies.currentEuroToRub).toFixed(4)}):Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{userCurrency:r.payload,exchangeCurrency_1:"USD",exchangeCurrency_2:"EUR",userSumInputMoney_1:(e.currentCurrencies.total*e.currentCurrencies.currentRubToUsd).toFixed(4),userSumInputMoney_2:(e.currentCurrencies.total*e.currentCurrencies.currentRubToEuro).toFixed(4)})}(e,r);case"UPDATE_CURRENCY":var n=r.payload,u=n.currentUsdToRub,c=n.currentEuroToRub,t=(u/c).toFixed(5),a=(c/u).toFixed(4),s=(1/c).toFixed(4),o=(1/u).toFixed(4);return Object(M.a)(Object(M.a)({},e.currentCurrencies),{},{currentUsdToRub:u,currentUsdToEuro:t,currentEuroToRub:c,currentEuroToUsd:a,currentRubToEuro:s,currentRubToUsd:o});default:return e.currentCurrencies}},w=function(e,r){return{currentCurrencies:N(e,r)}},F=n(28),k=Object(F.a)(w);t.a.render(Object(y.jsx)(j.a,{store:k,children:Object(y.jsx)(l.a,{children:Object(y.jsx)(I,{})})}),document.getElementById("root"))}},[[44,1,2]]]);
//# sourceMappingURL=main.26621f77.chunk.js.map