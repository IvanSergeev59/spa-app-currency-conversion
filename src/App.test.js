import React from "react";
import { render, screen } from "@testing-library/react";
import {UserCurrencyContainer} from "./components/user-currency/user-currency.js";

// check text UserCurrencyContainer component
test('UserCurrencyContainer renders Ваша валюта:', () => {
    const {getByText} = render(<UserCurrencyContainer />); 
    const linkElement = getByText(/Ваша валюта:/i);
    expect(linkElement).toBeInTheDocument();
  });

test('UserCurrencyContainer snapshot', () => {
    const {asFragment} = render(<UserCurrencyContainer />);
    expect(asFragment(<UserCurrencyContainer />)).toMatchSnapshot();
})