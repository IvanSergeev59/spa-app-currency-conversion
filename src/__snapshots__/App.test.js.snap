// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="currency-list currency-list-user"
  >
    <p
      class="currency-list-_p"
    >
      Ваша валюта:
    </p>
    <select
      aria-label="Default select example"
      class="form-select"
    >
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="EURO"
      >
        EURO
      </option>
      <option
        value="RUB"
      >
        RUB
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`UserCurrencyContainer snapshot 1`] = `
<DocumentFragment>
  <div
    class="currency-list currency-list-user"
  >
    <p
      class="currency-list-_p"
    >
      Ваша валюта:
    </p>
    <select
      aria-label="Default select example"
      class="form-select"
    >
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="EURO"
      >
        EURO
      </option>
      <option
        value="RUB"
      >
        RUB
      </option>
    </select>
  </div>
</DocumentFragment>
`;
